global/global_test.go

```diff
@@ -18,13 +18,199 @@
 package global
 
 import (
+	"bytes"
+	"crypto/rand"
+	"crypto/rsa"
+	"crypto/x509"
+	"crypto/x509/pkix"
+	"encoding/pem"
+	"fmt"
+	"io/ioutil"
+	"math/big"
 	"os"
 	"path/filepath"
```

Comment:
The `global.go` test coverage could be promoted if we introduce [mock-patch](https://github.com/bouk/monkey) to mock the system call.